jetty-websocket/websocket-common/src/main/java/org/eclipse/jetty/websocket/common/extensions/compress/CompressExtension.java

```diff
@@ -46,6 +46,11 @@
     protected static final byte[] TAIL_BYTES = new byte[]{0x00, 0x00, (byte)0xFF, (byte)0xFF};
     protected static final ByteBuffer TAIL_BYTES_BUF = ByteBuffer.wrap(TAIL_BYTES);
     private static final Logger LOG = Log.getLogger(CompressExtension.class);
+    
+    /**
+     * thread local accumulator
+     */
+    protected ThreadLocal<ByteAccumulator> accumulator = ThreadLocal.withInitial(() -> newByteAccumulator());
```

Comment:
Jetty's websocket implementation is asynchronous.

There's no guarantee that this CompressExtension instance will be used by the same Thread each time.
I'm -1 on this `ThreadLocal` addition.